pkg/kds/client/stream.go

```diff
@@ -40,11 +43,20 @@ func NewKDSStream(s mesh_proto.KumaDiscoveryService_StreamKumaResourcesClient, c
 }
 
 func (s *stream) DiscoveryRequest(resourceType model.ResourceType) error {
+	remoteCpVersionJSON, err := json.Marshal(kuma_version.Build)
+	if err != nil {
+		return err
+	}
 	return s.streamClient.Send(&envoy.DiscoveryRequest{
 		VersionInfo:   "",
 		ResponseNonce: "",
 		Node: &envoy_core.Node{
 			Id: s.clientId,
+			Metadata: &pstruct.Struct{
+				Fields: map[string]*pstruct.Value{
+					"version": {Kind: &pstruct.Value_StringValue{StringValue: string(remoteCpVersionJSON)}},
+				},
+			},
```

Comment:
I tested the code and it does not work outside of unit tests.
Here you put the version in StringValue, but in status tracker you try to extract the StructValue.

Please construct here proto KumaCpVersion that you created based on kuma_version.Build. Marshal this to StructValue.

I recommend testing it with kuma-demo multicluster universal
https://github.com/kumahq/kuma-demo/tree/multicluster/multicluster/universal

Reply:
tested with local universal setup and cp version is present in global cp.
<img width="883" alt="Screen Shot 2021-01-08 at 5 03 14 PM" src="https://user-images.githubusercontent.com/14926492/104012552-01cd1200-51d6-11eb-893b-344100e86df6.png">

Reply:
Ok, now it works 🆗